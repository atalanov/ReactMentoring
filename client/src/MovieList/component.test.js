import React from 'react';
import { shallow, render } from 'enzyme';
import Component from './component';

describe('Component tests', () => {
    test('Component renders correctly', () => {
        const component = shallow(<Component />);
        expect(component).toMatchSnapshot();
    });
});
