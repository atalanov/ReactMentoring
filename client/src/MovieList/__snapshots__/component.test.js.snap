// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList tests Component MovieList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          item={
            Object {
              "id": 0,
              "jenre": "Comedy",
              "name": "Jackie Brown",
              "posterUrl": "https://images-na.ssl-images-amazon.com/images/I/91e2jI9N9sL._SX355_.jpg",
              "year": 1997,
            }
          }
        />,
        <Unknown
          item={
            Object {
              "id": 1,
              "jenre": "Comedy",
              "name": "Reservoir Dogs",
              "posterUrl": "http://centuryclub.co.uk/wp-content/uploads/2017/10/reservoir_dogs_1992_8.jpg",
              "year": 1992,
            }
          }
        />,
        <Unknown
          item={
            Object {
              "id": 2,
              "jenre": "Comedy",
              "name": "Four Rooms",
              "posterUrl": "http://is2.mzstatic.com/image/thumb/Video5/v4/df/40/02/df400274-1155-7414-01e9-e7ff4f7f56f0/source/227x227bb.jpg",
              "year": 1995,
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "item": Object {
            "id": 0,
            "jenre": "Comedy",
            "name": "Jackie Brown",
            "posterUrl": "https://images-na.ssl-images-amazon.com/images/I/91e2jI9N9sL._SX355_.jpg",
            "year": 1997,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "item": Object {
            "id": 1,
            "jenre": "Comedy",
            "name": "Reservoir Dogs",
            "posterUrl": "http://centuryclub.co.uk/wp-content/uploads/2017/10/reservoir_dogs_1992_8.jpg",
            "year": 1992,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "item": Object {
            "id": 2,
            "jenre": "Comedy",
            "name": "Four Rooms",
            "posterUrl": "http://is2.mzstatic.com/image/thumb/Video5/v4/df/40/02/df400274-1155-7414-01e9-e7ff4f7f56f0/source/227x227bb.jpg",
            "year": 1995,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            item={
              Object {
                "id": 0,
                "jenre": "Comedy",
                "name": "Jackie Brown",
                "posterUrl": "https://images-na.ssl-images-amazon.com/images/I/91e2jI9N9sL._SX355_.jpg",
                "year": 1997,
              }
            }
          />,
          <Unknown
            item={
              Object {
                "id": 1,
                "jenre": "Comedy",
                "name": "Reservoir Dogs",
                "posterUrl": "http://centuryclub.co.uk/wp-content/uploads/2017/10/reservoir_dogs_1992_8.jpg",
                "year": 1992,
              }
            }
          />,
          <Unknown
            item={
              Object {
                "id": 2,
                "jenre": "Comedy",
                "name": "Four Rooms",
                "posterUrl": "http://is2.mzstatic.com/image/thumb/Video5/v4/df/40/02/df400274-1155-7414-01e9-e7ff4f7f56f0/source/227x227bb.jpg",
                "year": 1995,
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "id": 0,
              "jenre": "Comedy",
              "name": "Jackie Brown",
              "posterUrl": "https://images-na.ssl-images-amazon.com/images/I/91e2jI9N9sL._SX355_.jpg",
              "year": 1997,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "id": 1,
              "jenre": "Comedy",
              "name": "Reservoir Dogs",
              "posterUrl": "http://centuryclub.co.uk/wp-content/uploads/2017/10/reservoir_dogs_1992_8.jpg",
              "year": 1992,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "id": 2,
              "jenre": "Comedy",
              "name": "Four Rooms",
              "posterUrl": "http://is2.mzstatic.com/image/thumb/Video5/v4/df/40/02/df400274-1155-7414-01e9-e7ff4f7f56f0/source/227x227bb.jpg",
              "year": 1995,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
